micrometer-core/src/test/java/io/micrometer/core/instrument/binder/hystrix/MicrometerMetricsPublisherCommandTest.java

```diff
@@ -71,16 +74,24 @@ void testCumulativeCounters() throws Exception {
             new SuccessCommand(key).execute();
         }
 
-        Thread.sleep(500);
-
         List<Tag> tags = Tags.zip("group", "MicrometerGROUP", "key", "MicrometerCOMMAND-A");
-        assertThat(registry.find("hystrix.execution").tags(tags).tags("event", "success").functionCounter().map(FunctionCounter::count)).hasValue(40.0);
-        assertThat(registry.find("hystrix.execution").tags(tags).tags("event", "timeout").functionCounter().map(FunctionCounter::count)).hasValue(5.0);
-        assertThat(registry.find("hystrix.execution").tags(tags).tags("event", "failure").functionCounter().map(FunctionCounter::count)).hasValue(10.0);
+
+        assertExecutionMetric(registry, "success", 24.0);
+        assertThat(registry.find("hystrix.execution").tags(tags).tags("event", "timeout").functionCounter().map(FunctionCounter::count)).hasValue(3.0);
+        assertThat(registry.find("hystrix.execution").tags(tags).tags("event", "failure").functionCounter().map(FunctionCounter::count)).hasValue(6.0);
         assertThat(registry.find("hystrix.execution").tags(tags).tags("event", "short_circuited").functionCounter().map(FunctionCounter::count)).hasValue(0.0);
         assertThat(registry.find("hystrix.circuit.breaker.open").tags(tags).gauge().map(Gauge::value)).hasValue(0.0);
     }
 
+    private void assertExecutionMetric(SimpleMeterRegistry registry, String eventType, double count) {
+        Flux.interval(Duration.ofMillis(50))
```

Comment:
@checketts Polished this up a bit. If I understand the logic in the previous iteration, it would have looped indefinitely if the condition was never met.

Reply:
Also, pretty good example of `Flux.interval` ;)

Reply:
It would have looped indefinitely? I had it check `while(System.currentTimeMillis() - startTime < waitTimeMs) {` so it should have exited correctly. 

Thanks for the `Flux.interval`! The more I see examples like this the more hopeful I am that I'll make the reactive jump eventually. ;)

Reply:
Hmm, one drawback with the flux, is we lost some details on failure. Like the build just failed, but it isn't clear why. 30 seconds should have been plenty for the metric to persist.